### DISCUSSION TITLE
Autocomplete - Customizing the display value?

### DISCUSSION THREAD

**Discussion by ghost** [mantinedev/mantine#1133] [Q&A] [answered]
I have an endpoint that I am reaching out to fetch an array of objects and then I am rendering those in the autocomplete. Everything is working great there! For the value, I need to grab the ID from the object to assign to a form field. However, if I do this then it shows the number in the field and attempts to query by that ID instead of the text value.

Is there a way to customize what is shown as the value in the text field when an item is select?

**Comment by rtivital** [comment, marked as answer]
Yes, you can provide your own item component – https://mantine.dev/core/autocomplete/#custom-item-component

**Reply by ghost** [reply]
@rtivital Correct and I have that in place for the custom options that you can choose from. Once you select an option, it converts back to the value. Example - My select option says "Florida", but when I select it it changes to the value I need to send back to the server, which is "75".

**Reply by rtivital** [reply]
Autocomplete does not support custom labels, you can use Select instead if you need this.

**Reply by ghost** [reply]
I wondered if that was the case - thank you for the confirmation. Looking at the Select component, is that where I need to use the [Createable](https://mantine.dev/core/select/#creatable) to customize what shows with the `getCreateLabel`? The example doesn't seem to show the "+ Create" text so I just wanted to make sure!

**Reply by jayasreek96** [reply]
use Select Component from mantine .It has label and value and render options using itemComponent and in OnChange props you can get value . here is the code
interface ItemProps extends SelectItemProps {
  _id: any
  name: `string`
}
// here storing array of objects with label and value
const store = data.map((item: any) => ({
    ...item,
    value: item.id,
    label: item.anyValue,
  }))

  // eslint-disable-next-line react/display-name
  const SelectItem = forwardRef<HTMLDivElement, ItemProps>(
    ({ value, label, name, ...others }: ItemProps, ref) => (
      <div ref={ref} {...others}>
        <Group noWrap>
          <div>
            <Text size="sm">{label}</Text>
          </div>
        </Group>
      </div>
    )
  )

 <Select
              label="Select Brand"
              placeholder="Select Brand"
              itemComponent={SelectItem}
              data={brand}
              withAsterisk
              maxDropdownHeight={400}
              nothingFound="Nobody here"
              onChange={(value: any) => {
                console.log('val8', value)
                formik.setFieldValue('brandId', value)
              }}
              styles={(theme) => ({
                item: {
                  // applies styles to selected item
                  '&[data-selected]': {
                    '&, &:hover': {
                      backgroundColor:
                        theme.colorScheme === 'dark'
                          ? theme.colors.teal[9]
                          : theme.colors.teal[1],
                      color:
                        theme.colorScheme === 'dark'
                          ? theme.white
                          : theme.colors.teal[9],
                    },
                  },

                  // applies styles to hovered item (with mouse or keyboard)
                  '&[data-hovered]': {},
                },
              })}
            />
```
